task-sdk/src/airflow/sdk/log.py

```diff
@@ -116,6 +118,37 @@ def drop_positional_args(logger: Any, method_name: Any, event_dict: EventDict) -
     return event_dict
 
 
+def custom_format(logger: Any, method_name: Any, event_dict: EventDict) -> EventDict:
+    record = event_dict.get("_record", None)
+    func_name = event_dict.pop("func_name", "")
+    warning_lineno = event_dict.pop("warning_lineno", "")
+    lineno = event_dict.pop("lineno", 0)
+    lineno = warning_lineno or lineno
+    warning_pathname = event_dict.pop("warning_pathname", "")
+    pathname = event_dict.pop("pathname", "")
+    pathname = warning_pathname or pathname
+    # Ignore the logs forwarded from subprocess, otherwise we got wrong callsite
+    if (chan := event_dict.get("chan", None)) and chan == "logs":
+        event_dict.pop("chan")
+        return event_dict
+    if not record:
+        # Build LogRecord to format log for stderr, stdout and warnings
+        record = logging.LogRecord(
+            name=event_dict.get("logger", ""),
+            level=structlog.stdlib.NAME_TO_LEVEL[event_dict.get("level", "")],
+            pathname=pathname,
+            lineno=lineno,
+            msg=event_dict.get("event", ""),
+            args=event_dict.pop("positional_args", ()),
+            exc_info=event_dict.pop("exc_info", None),
+            func=func_name,
+        )
+    record.msg = event_dict.pop("event", "")
+    formatter = logging.Formatter(settings.LOG_FORMAT)
```

Comment:
A number of things need changing here:

1. Creating a new formatter for every single stdlib log message received is going to massively hurt performance
2. we want everything as structured JSON fields, not plain text. So we can't us stdlib formatters really.
3. This entire function is complicated and hard to work out what it's actually doing?  I wrote all the rest of the logging code and _I_ can't work it out so we need to refactor this so the intent is clearer. What does this fn do?